Add: 필터 기능용 컴포넌트 파일 생성

lemongucci opened the pull request:
- 반응형 구현
- 향후 HomeList 페이지 상하단에 붙일 Nav, Footer을 고려한 최종 레이아웃 완료
- List 부분 안에 Card 컴포넌트 넣을 예정
- 섹션별 Styled Component의 이름을 수정할 예정

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 버튼 활성화 기능 구현(예시)
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화(예시)

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [ ] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [ ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ ] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.
- [ ] Git rebase와 squash를 했고 커밋 수가 하나 인것을 확인 했습니다.

saengmotmi commented:
@lemongucci 현재 리뷰 요청한 상태인가요? 라벨 달아서 PR 상태 체크해주세요

yojuyoon commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,88 @@
+  background-color: black;
+  opacity: 0.3;
+  height: 50px;
+  width: 840px;
```
css속성 순서는 아래와 같습니다!
- Layout Properties (position, float, clear, display)
- Box Model Properties (width, height, margin, padding)
- Visual Properties (color, background, border, box-shadow)
- Typography Properties (font-size, font-family, text-align, text-transform)
- Misc Properties (cursor, overflow, z-index)

yojuyoon commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,88 @@
+  height: 178px;
+`;
+
+//함수
```
styled-component를 컴포넌트 상단에 설정해줄 수도 있지만 저희 컨벤션으로는 하단에 설정해주고 있습니다! 🙂

lemongucci replied:
네! 알겠습니다!!

yojuyoon commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,88 @@
+          <Gmap />
+        </div>
+        <div>
+          <Pagination></Pagination>
```
self-closing tag 가능합니다!

yojuyoon commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,88 @@
 const HomeList = () => {
-  return <div>HomeList</div>;
+  return (
+    <div>
```
빈 fragment로 감싸줄 수 있겠네요!

lemongucci replied:
감사합니다^^

yojuyoon commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,88 @@
+          <Pagination></Pagination>
+        </div>
+      </Table>
+      <Footer></Footer>
```
self-closing tag!

yojuyoon reviewed (commented):
양효님! 리뷰 반영 부탁드리고 기능 구현에도 힘써주세요! 💪🏻🔥 다음주도 화이팅!!

Commit:
Add: Filter 용 모달 컴포넌트 파일 생성
- 백엔드 사이드와 어느 정도 수준으로 필터 기능을 구현할 지 상의한 후
필터 기능 구현할 예정
- Google-map-react 라이브러리 사용하여 우측 지도 컴포넌트 생성 후 백엔드
사이드와 기능 구현 수준 상의할 예정

claire-euni commented on src/Pages/HomeList/Components/Card.js:
```diff
@@ -0,0 +1,206 @@
+import React from "react";
+import styled from "styled-components";
+
+const CardWrapper = styled.div`
```
저희는 컨벤션으로 스타일 컴포넌트는 return문 아래 하단에 작성하고 있습니다. 위치 수정해주세요~

lemongucci replied:
예 알겠습니다!

claire-euni commented on src/Pages/HomeList/Components/Card.js:
```diff
@@ -0,0 +1,206 @@
+const Card = () => {
+  return (
+    <CardWrapper>
+      <div>
+        <div>div 200 300</div>
+        <div>
+          <div>
+            <div>
+              <p>Entire House in Tapdong-ro, Jeju</p>
+              <div>
+                <h4>
+                  한달살기 주택 제주시 공항과 가까운 해변 102에 가서 신나게
+                  놀다보면 벌써 아침, 밥을 먹고 또 신나게 놀다보면 벌써 저녁
+                </h4>
+              </div>
+            </div>
```
div들의 파티인가요..........보다는 semantic tag사용 혹은 스타일 컴포넌트로 분리하는게 가독성에 좋을것 같습니다.
그리고 이렇게 많은 div가 필요한지 고민해보세요~

lemongucci replied:
Css에 대한 지식 부족으로 인해ㅠㅠㅜㅠdiv 남용하게 된 거 같아요

claire-euni commented on src/Pages/HomeList/Components/Filters/CancellationFlexibility.js:
```diff
@@ -0,0 +1,185 @@
+        flex-direction: row;
+        align-items: center;
+        justify-content: space-between;
+        & > button:first-child {
+          background-color: transparent;
+          border: none;
+          border-bottom: 1.5px solid rgb(235, 235, 235);
+          outline: none;
+          cursor: pointer;
+          color: rgb(235, 235, 235);
+          font-size: 16px;
+          font-weight: 600;
+          height: 20px;
+          padding: 0;
```
- 지난 리뷰 확인해서 전체적으로 css 순서 수정해주세요. 어디에서는 width,height이 먼저 시작되고 어디에서는 position이 먼저시작하네요. 통일성 있게 작성해주세요~
- 네스팅 되어있으므로 & > 가 필요하지 않을것 같습니다.

lemongucci replied:
네 알겠습니다 감사합니다^^

claire-euni commented on src/Pages/HomeList/Components/Filters/InstantBook.js:
```diff
@@ -0,0 +1,185 @@
+        -webkit-transition: left 0.25s ease;
+        transition: left 0.25s ease;
+      }
+      .switch-input:checked + .switch-label::before {
```
+는 무엇인가요?

lemongucci replied:
+앞에 선택자 바로 뒤에 붙는 태그들 중
+뒤에 오는 selector 에 스타일이나 속성을 주고 싶을 때 사용하는
selector라고 알고 있습니다

claire-euni commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,262 @@
 import React from "react";
+import styled, { keyframes } from "styled-components";
 
+import Card from "./Components/Card";
+import GoogleMap from "./Components/GoogleMap";
+
```
import 사이에 엔터는 불필요합니다 삭제해주세요~

lemongucci replied:
네!!

claire-euni commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,262 @@
-  return <div>HomeList</div>;
+  return (
+    <>
+      {/* 필터리스트 */}
+      {/* <MoreFilters /> */}
```
주석 부분은 항상 삭제해주세요 :)

lemongucci replied:
아직 없으면 헷갈려서 나중에 지우겠습니다

claire-euni commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,262 @@
+        <section>
+          <div>haha</div>
+        </section>
+
```
엔터 불필요!

claire-euni commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,262 @@
+                      </div> */}
+                    </div>
+
+                    <div>
+                      <div className="btnFilter">
+                        <Button>
+                          <span>즉시 예약</span>
+                        </Button>
+                      </div>
+                      {/* <div className="modalFilter">
+                        <InstantBook />
+                      </div> */}
+                    </div>
```
이 부분이 반복되고 있는데 map()을 활용해볼 수 있겠네요!

lemongucci replied:
네!! 이 부분과 관련해서 나중에 질문하러 가도 될까요???

claire-euni commented on src/Pages/HomeList/HomeList.js:
```diff
@@ -1,7 +1,262 @@
+              </div>
+
+              <div>
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
+                <Card />
```
동일한 내용을 담은 컴포넌트가 이렇게 많이 필요한건가요? for문을 사용해볼 수 있겠습니다

lemongucci replied:
네 감사합니다!!

claire-euni reviewed (commented):
지난 리뷰와 이번 리뷰 반영해서 리팩토링 진행해주세요 양효님 🙌

- 전체적으로 css 순서 통일감있게 수정해주세요
- div 사용이 너무 많은데 class명도 없어서 스타일을 줄 때 뚜렷한 구분이 되지 않습니다. semantic tag나 스타일 컴포넌트로 분리하는 연습해보세요~
